modules/RTC/TPCUtils.js

```diff
@@ -418,7 +417,7 @@ export class TPCUtils {
                         parameters.encodings.forEach(encoding => {
                             encoding.active = true;
                         });
-                        transceiver.sender.setParameters(parameters);
+                        setParamPromises.push(transceiver.sender.setParameters(parameters));
```

Comment:
Was this the reason for the onnegotiationneeded loop ? I somehow missed the fact that RTCRtpSender.setParameters is async :(

Reply:
No worries. It wasn't - the onnegotiationneeded loop was caused on my experimental branch where I was setting sendrecv direction on the recvonly transceiver for the remote stream.